Add function to load YAML files

tobiasge opened the pull request:
This commit starts to remove some code redundancy from the startup
scripts for easier maintenance.

Related Issue:

## New Behavior
- Use shared functions in startup scripts

## Contrast to Current Behavior
- The same code is use in multible files

## Discussion: Benefits and Drawbacks
- Maintenance of script should be easier

## Changes to the Wiki
- None

## Proposed Release Note Entry
- Internal code maintenance: Started to extract redundant code into includes

## Double Check
* [x] I have read the comments and followed the PR template.
* [x] I have explained my PR according to the information in the comments.
* [x] My PR targets the `develop` branch.

tobiasge committed:
Add function to load YAML files

This commit starts to remove some code redundancy from the startup
scripts for easier maintenance.

cimnine commented on startup_scripts/includes/__init__.py:
```diff
@@ -0,0 +1,11 @@
+
+
+def load_yaml(yaml_file: str):
+    yf = Path(yaml_file)
```
I believe that we should stick to one indention style. You suggested to change the indention to four spaces in private. Until we've done that everywhere, I believe we should stick to the status quo which is two spaces.

cimnine commented on startup_scripts/includes/__init__.py:
```diff
@@ -0,0 +1,11 @@
+from ruamel.yaml import YAML
+from pathlib import Path
+
+
```
Although there are other places where there are two empty lines, to me they're a waste of screen estate most of the time.

cimnine commented on startup_scripts/260_ip_addresses.py:
```diff
@@ -3,63 +3,56 @@
-  ip_addresses = yaml.load(stream)
+ip_addresses = load_yaml('/opt/netbox/initializers/ip_addresses.yml')
 
+if ip_addresses is not None:
```
Maybe we can turn this `if` around:

```suggestion
if ip_addresses is None:
  sys.exit()
```

That would prevent a whole level of indention.

cimnine reviewed (changes requested):
That's already a huge improvement as it is, thanks Tobias.

Some thought about how I think this could become even better:

- I don't like the module's name `includes` as it is very generic. I would suggest `startup_script_utils` or something in that direction.
- I think we should put the `def load_yaml()` into it's own file, e.g. `load.py`. Then, `__init__.py` would [import that file](https://docs.python.org/3/reference/import.html#submodules):
  ```python
  from .load import load_yaml`.
  ```
  This avoids the mistake of the past so that the `__init__.py` file does not grow into a big pile of code.
- Maybe you can add the `.editorconfig` here already, with python code indention being two spaces.

tobiasge commented:
> * I don't like the module's name `includes` as it is very generic. I would suggest `startup_script_utils` or something in that direction.

- Renamed

> * I think we should put the `def load_yaml()` into it's own file, e.g. `load.py`. Then, `__init__.py` would [import that file](https://docs.python.org/3/reference/import.html#submodules):
>   ```python
>   from .load import load_yaml`.
>   ```
>   This avoids the mistake of the past so that the `__init__.py` file does not grow into a big pile of code.

I had that thought at the beginning, but didn't implement it, because I added only one function. 
But I think you are right. We should do this correct from the beginning.

cimnine reviewed (approved):
This is such a big improvement! 😻 Thank you Tobias!
